### PR TITLE
Explain out-of-the-box alerts

### PR DESCRIPTION
**Pull request by catherineluse**
This PR corresponds to this issue https://github.com/rancher/docs/issues/1565

@loganhz , I have a few questions.

- what is a deployment event?
- why would someone want an alert when a deployment event happens?
- When you go to a project in Rancher and go to the alerts page, it shows an alert about memory quota. Is the memory quota set at the project level?

### CODE REVIEW

**Comment by loganhz** [comment]
@aiwantaozi

**Comment by aiwantaozi** [comment]
Hi @catherineluse 
Re 1: Kubernetes events are objects that provide insight into what is happening inside a cluster, such as what decisions were made by scheduler or why some pods were evicted from the node. Deployment event is for resource depoyment. You could get all events by ```kubectl get event --all-namespaces```. In the rancher UI, you could easily see the event under workloads.

image

Re 2:  For example, there are normal events and warning events, user could configure to be got alert when warning event happened.

Re 3:  It is the quota you set for workload, you could see it under worklaod `Security & Host Config` tag.

File: content/rancher/v2.x/en/cluster-admin/tools/alerts/default-alerts/_index.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+Etcd is the key-value store that contains the state of the Kubernetes cluster. If you provide monitoring on your cluster, Rancher provides default alerts if the monitoring detects a potential problem with etcd.
+
+A leader is the node that handles all client requests that need cluster consensus. For more information, you can refer to this [explanation of how etcd works.](https://rancher.com/blog/2019/2019-01-29-what-is-etcd/#how-does-etcd-work)
+
+The leader of the cluster can change in response to certain events. It is normal for the leader to change, but too many changes can indicate a problem with the network or a high CPU load. With longer latencies, the default etcd configuration may cause frequent heartbeat timeouts, which trigger a new leader election.
+
+| Alert | Explanation |
+|-------|-------------|
+| A high number of leader changes within the etcd cluster are happening | A warning alert is triggered when the leader changes more than three times in one hour. |
+| Database usage close to the quota 500M | A warning alert is triggered when the size of etcd exceeds 524.288M.|
```
Comment (aiwantaozi) [inline content/rancher/v2.x/en/cluster-admin/tools/alerts/default-alerts/_index.md]: exceeds 500M. 524288000 bytes = 500M

File: content/rancher/v2.x/en/cluster-admin/tools/alerts/default-alerts/_index.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+| Etcd is unavailable | A critical alert is triggered when etcd becomes unavailable. |
+| Etcd member has no leader | A critical alert is triggered when etcd does not have a leader for at least three minutes. |
+
```
Comment (aiwantaozi) [inline content/rancher/v2.x/en/cluster-admin/tools/alerts/default-alerts/_index.md]: A critical alert is triggered when etcd cluster does not have a leader for at least three minutes.

File: content/rancher/v2.x/en/cluster-admin/tools/alerts/default-alerts/_index.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+| High node memory utilization | A warning alert is triggered if the node uses more than 80 percent of its available memory for at least three minutes. |
+| Node disk is running full within 24 hours | A critical alert is triggered if the disk space on the node is expected to run out in the next 24 hours based on the disk growth over the last 6 hours. |
+
+# Project-level Alerts
+When you enable monitoring for the project, some project-level alerts are provided.
+
+| Alert | Explanation |
+|-------|-------------|
+| Less than half workload available | A critical alert is triggered if less than half of a workload is available, based on workloads where the key is `app` and the value is `workload`. |
+| Memory usage close to the quota | A warning alert is triggered if the project's memory usage exceeds the memory resource quota that is set for the workload. You can see the memory limit in the Rancher UI if you go to the workload under the **Security & Host Config** tab. |
```
Comment (aiwantaozi) [inline content/rancher/v2.x/en/cluster-admin/tools/alerts/default-alerts/_index.md]: A warning alert is triggered if workload's memory usage exceeds the memory resource quota is set for the workload. You can see the memory limit in the Rancher UI if you go to the workload under the **Security & Host Config** tab.

File: content/rancher/v2.x/en/cluster-admin/tools/_index.md [hunk 4]
```diff
@@ -23,6 +23,8 @@ Notifiers and alerts are two features that work together to inform you of events
 
 Notifiers and alerts are built on top of the [Prometheus Alertmanager](https://prometheus.io/docs/alerting/alertmanager/). Leveraging these tools, Rancher can notify [cluster owners]({{< baseurl >}}/rancher/v2.x/en/admin-settings/rbac/cluster-project-roles/#cluster-roles) and [project owners]({{< baseurl >}}/rancher/v2.x/en/admin-settings/rbac/cluster-project-roles/#project-roles) of events they need to address.
 
+When you create a cluster and enable monitoring for the cluster, some alerts are added by default. For details, refer to the [documentation on default alerts.]({{< baseurl >}}/rancher/v2.x/en/cluster-admin/tools/alerts/default-alerts)
+
```
Comment (aiwantaozi) [inline content/rancher/v2.x/en/cluster-admin/tools/_index.md]: When you create a cluster, there are some predefined alerts. For details, refer to the [documentation on default alerts.]({{< baseurl >}}/rancher/v2.x/en/cluster-admin/tools/alerts/default-alerts)

File: content/rancher/v2.x/en/cluster-admin/tools/alerts/default-alerts/_index.md (hunk 1 above)
Comment (catherineluse) [reply content/rancher/v2.x/en/cluster-admin/tools/alerts/default-alerts/_index.md]: fixed

File: content/rancher/v2.x/en/cluster-admin/tools/alerts/default-alerts/_index.md (hunk 2 above)
Comment (catherineluse) [reply content/rancher/v2.x/en/cluster-admin/tools/alerts/default-alerts/_index.md]: fixed

File: content/rancher/v2.x/en/cluster-admin/tools/alerts/default-alerts/_index.md (hunk 3 above)
Comment (catherineluse) [reply content/rancher/v2.x/en/cluster-admin/tools/alerts/default-alerts/_index.md]: fixed

File: content/rancher/v2.x/en/cluster-admin/tools/_index.md (hunk 4 above)
Comment (catherineluse) [reply content/rancher/v2.x/en/cluster-admin/tools/_index.md]: fixed

Comment (aiwantaozi) [reply content/rancher/v2.x/en/cluster-admin/tools/_index.md]: Hi, enable monitoring isn't required after created the cluster, predefined alert rules get created.

Comment (catherineluse) [reply content/rancher/v2.x/en/cluster-admin/tools/_index.md]: Oops, I thought I changed that. now it's fixed

**Comment by aiwantaozi** [comment]
LGTM

File: content/rancher/v2.x/en/cluster-admin/tools/alerts/default-alerts/_index.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+weight: 1
+---
+
+When you create a cluster, some alert rules are predefined. These alerts notify you about signs that the cluster could be unhealthy. You can receive these alerts if you configure a [notifier]({{<baseurl>}}/rancher/v2.x/en/cluster-admin/tools/notifiers) for them.
+
+Several of the alerts use Prometheus expressions as the metric that triggers the alert. For more information on how expressions work, you can refer to the Rancher [documentation about Prometheus expressions]({{< baseurl >}}
+/rancher/v2.x/en/cluster-admin/tools/monitoring/expression/) or the Prometheus [documentation about querying metrics](https://prometheus.io/docs/prometheus/latest/querying/basics/).
+
+# Alerts for etcd
+Etcd is the key-value store that contains the state of the Kubernetes cluster. If you provide monitoring on your cluster, Rancher provides default alerts if the monitoring detects a potential problem with etcd.
```
Comment (deniseschannon) [inline content/rancher/v2.x/en/cluster-admin/tools/alerts/default-alerts/_index.md]: Is "providing monitoring" the same as enabling monitoring? Or is it available for any monitoring?

File: content/rancher/v2.x/en/cluster-admin/tools/alerts/default-alerts/_index.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+| Alert | Explanation |
+|-------|-------------|
+| A high number of leader changes within the etcd cluster are happening | A warning alert is triggered when the leader changes more than three times in one hour. |
+| Database usage close to the quota 500M | A warning alert is triggered when the size of etcd exceeds 500M.|
+| Etcd is unavailable | A critical alert is triggered when etcd becomes unavailable. |
+| Etcd member has no leader | A critical alert is triggered when the etcd cluster does not have a leader for at least three minutes. |
+
+
+# Alerts for Kube Components
```
Comment (deniseschannon) [inline content/rancher/v2.x/en/cluster-admin/tools/alerts/default-alerts/_index.md]: `Alerts for Kubernetes Components`

File: content/rancher/v2.x/en/cluster-admin/tools/alerts/default-alerts/_index.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+# Alerts for Events
+Kubernetes events are objects that provide insight into what is happening inside a cluster, such as what decisions were made by the scheduler or why some pods were evicted from the node. In the Rancher UI, from the project view, you can see events for each workload.
+
+| Alert | Explanation |
+|-------|-------------|
+| Get warning deployment event | A warning alert is triggered by a warning event happens on a deployment. |
+
+
+# Alerts for Nodes
```
Comment (deniseschannon) [inline content/rancher/v2.x/en/cluster-admin/tools/alerts/default-alerts/_index.md]: In all the other sections, we describe the component (etcd, k8s components, events).

Maybe we should add a small note about what the node is.

File: content/rancher/v2.x/en/cluster-admin/tools/alerts/default-alerts/_index.md (hunk 5 above)
Comment (catherineluse) [reply content/rancher/v2.x/en/cluster-admin/tools/alerts/default-alerts/_index.md]: This is my mistake. Michela told me that the default alerts are included whether you enable monitoring or not. 

Changing it to:

> Rancher provides default alerts if the built-in monitoring detects a potential problem with etcd.

File: content/rancher/v2.x/en/cluster-admin/tools/alerts/default-alerts/_index.md (hunk 6 above)
Comment (catherineluse) [reply content/rancher/v2.x/en/cluster-admin/tools/alerts/default-alerts/_index.md]: changed heading

File: content/rancher/v2.x/en/cluster-admin/tools/alerts/default-alerts/_index.md (hunk 7 above)
Comment (catherineluse) [reply content/rancher/v2.x/en/cluster-admin/tools/alerts/default-alerts/_index.md]: explained what node is and linked to other doc that explains it more

**Pull request merged**